Lib/Allocator.hpp

```diff
@@ -35,6 +35,7 @@
 namespace Lib {
 // attempt to set a memory limit for this process by system call
 void setMemoryLimit(size_t bytes);
+long peakMemoryUsage();
```

Comment:
Maybe this should have a comment indicating that it is not clear what the unit is? Or move the Mac-specific thing inside this function?